Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,213 @@
+                geometry : geometry,
+                attributeLocations : {
+                    position : 0,
+                    textureCoordinates : 1
+                },
+                bufferUsage : BufferUsage.STATIC_DRAW,
+                interleave : true
+            });
+
+            computeEngine._viewportQuadVertexArray = vertexArray;
+        }
+
+        return vertexArray;
+    }
+
+    function createFramebuffer(context, texture) {
+        var fbo = new Framebuffer({
+            context : context,
+            colorTextures : [texture]
+        });
+        fbo.destroyAttachments = false;
+        return fbo;
+    }
+
+    function createViewportQuadShader(context, fragmentShaderSource) {
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : ViewportQuadVS,
+            fragmentShaderSource : fragmentShaderSource,
+            attributeLocations : viewportQuadAttributeLocations
+        });
+    }
+
+    function createRenderState(width, height) {
+        if ((!defined(renderStateScratch)) ||
+            (renderStateScratch.viewport.width !== width) ||
+            (renderStateScratch.viewport.height !== height)) {
+
+            renderStateScratch = RenderState.fromCache({
+                viewport : new BoundingRectangle(0, 0, width, height)
+            });
+        }
+        return renderStateScratch;
+    }
+
+    ComputeEngine.prototype.execute = function(computeCommand) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(computeCommand)) {
+            throw new DeveloperError('computeCommand is required.');
+        }
+        //>>includeEnd('debug');
+
+        if (defined(computeCommand.preExecutionCallback)) {
+            computeCommand.preExecutionCallback(computeCommand);
+        }
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(computeCommand.fragmentShaderSource) && !defined(computeCommand.shaderProgram)) {
+            throw new DeveloperError('computeCommand.fragmentShaderSource or computeCommand.shaderProgram is required.');
+        }
+
+        if (!defined(computeCommand.outputTexture)) {
+            throw new DeveloperError('computeCommand.outputTexture is required.');
+        }
+        //>>includeEnd('debug');
+
+        var texture = computeCommand.outputTexture;
+        var width = texture.width;
+        var height = texture.height;
+
+        var context = this._context;
+        var vertexArray = defined(computeCommand.vertexArray) ? computeCommand.vertexArray : getViewportQuadVertexArray(this);
+        var shaderProgram = defined(computeCommand.shaderProgram) ? computeCommand.shaderProgram : createViewportQuadShader(context, computeCommand.fragmentShaderSource);
+        var framebuffer = createFramebuffer(context, texture);
+        var renderState = createRenderState(width, height);
+        var uniformMap = defaultValue(computeCommand.uniformMap, defaultValue.EMPTY_OBJECT);
```

Comment:
I'm pretty sure it is OK for `uniformMap` to be `undefined` so just
```
var uniformMap = computeCommand.uniformMap;
```
should be fine.